### PR TITLE
Document template vars

### LINKED ISSUE #215: Document Template Vars
- Review of how empy and jinja2 are currently documented.
- Ensure that references to `CYLC_TEMPLATE_VARS` are included in the new documentation.

### PR DESCRIPTION
**Pull request by wxtim**
Closes #215 

- [x] Add information about CYLC_TEMPLATE_VARS.
- [x] Clarified precendence of `-s` `--set-file` and plugin template vars.

Deliberately not done:
Removed jinja2 references from documentation which should be template variable agnostic.

### CODE REVIEW

File: src/user-guide/writing-workflows/jinja2.rst [hunk 1]
```diff
@@ -348,6 +360,40 @@ will show the workflow with the Jinja2 variables as set.
...
+      initial cycle point = {{ICP}}
+      [[dependencies]]
+         P1Y = Task1
+
+
+.. code-block:: shell
+
+   > cylc config . -i "[scheduling]initial cycle point"
+   1068
+
```
Comment (datamel) [inline src/user-guide/writing-workflows/jinja2.rst:394]: Not sure what I'm doing wrong here @wxtim, hopefully you can spot my mistake...I've tried out this example but hitting an error...I'm on your branches for https://github.com/cylc/cylc-flow/pull/4360 and the cylc-rose sibling, having pip installed them both and I'm getting.... 
image

Comment (wxtim) [reply src/user-guide/writing-workflows/jinja2.rst:394]: I don't think that rose supports spaces in `key = value`. Try `ICP=1068`

File: src/user-guide/writing-workflows/jinja2.rst [hunk 2]
```diff
@@ -348,6 +360,40 @@ will show the workflow with the Jinja2 variables as set.
...
+add a ``rose-suite.conf`` file which contains a ``[template variables]``
+section which the plugin makes available to Cylc:
+
+For example, with Cylc Rose installed:
+
+.. code-block:: ini
+
+   # rose-suite.conf
+   [template variables]
+   ICP = 1068
```
Comment (wxtim) [inline src/user-guide/writing-workflows/jinja2.rst:377]:
```suggestion
ICP=1067
```

File: src/user-guide/writing-workflows/jinja2.rst (hunk 1 above)
Comment (datamel) [reply src/user-guide/writing-workflows/jinja2.rst:394]: I tried without the spaces :( No luck, same error. My suspicion is it is my environment, I will try on a fresh one and see if that helps.

Comment (wxtim) [reply src/user-guide/writing-workflows/jinja2.rst:394]: :(

**Review by datamel — APPROVED** [review]
Failing tutorial unrelated. I have built and read the docs, including working through the example.
Looks great, thanks @wxtim.

File: src/user-guide/writing-workflows/jinja2.rst [hunk 3]
```diff
@@ -239,13 +241,23 @@ Here's the result:
...
+You can provide template variables to Cylc in 3 ways:
 
-This can be done on a case-by-case basis using the ``-s`` option e.g:
+- Using the ``--set-file`` option.
+- Using the ``-s`` option.
+- Using a plugin, such as :ref:`Cylc Rose`.
+
+.. note::
+
+   If the same variable is set by more than one method the last source in the
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/jinja2.rst:255]:
```suggestion
   If the same variable is set by more than one method, the last source in the
```

File: src/user-guide/writing-workflows/jinja2.rst [hunk 4]
```diff
@@ -239,13 +241,23 @@ Here's the result:
...
+--------------------------------------------
 
-The values of Jinja2 variables can be passed in from the Cylc command
-line rather than hardwired in the workflow configuration.
+You can provide template variables to Cylc in 3 ways:
 
-This can be done on a case-by-case basis using the ``-s`` option e.g:
+- Using the ``--set-file`` option.
+- Using the ``-s`` option.
+- Using a plugin, such as :ref:`Cylc Rose`.
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/jinja2.rst:251]: Would be nice if these could link to the sections below. But not essential

File: src/user-guide/writing-workflows/jinja2.rst [hunk 5]
```diff
@@ -348,6 +360,40 @@ will show the workflow with the Jinja2 variables as set.
...
 
+Using a plugin
+^^^^^^^^^^^^^^
+
+Template plugins such as :ref:`Cylc Rose` should provide a set of template
+variables which can be provided to Cylc. For example, using Cylc Rose you
+add a ``rose-suite.conf`` file which contains a ``[template variables]``
+section which the plugin makes available to Cylc:
+
+For example, with Cylc Rose installed:
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/jinja2.rst:371]:
```suggestion
add a ``rose-suite.conf`` file containing a ``[template variables]``
section which the plugin makes available to Cylc:
```

File: src/user-guide/writing-workflows/jinja2.rst [hunk 6]
```diff
@@ -348,6 +360,40 @@ will show the workflow with the Jinja2 variables as set.
...
+
+.. code-block:: ini
+
+   # rose-suite.conf
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/jinja2.rst:375]:
```suggestion
.. code-block:: ini
   :caption: rose-suite.conf
   
```

File: src/user-guide/writing-workflows/jinja2.rst [hunk 7]
```diff
@@ -348,6 +360,40 @@ will show the workflow with the Jinja2 variables as set.
...
+   [template variables]
+   ICP=1068
+
+.. code-block:: cylc
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/jinja2.rst:379]:
```suggestion
.. code-block:: cylc
   :caption: flow.cylc
```

File: src/user-guide/writing-workflows/jinja2.rst (hunk 4 above)
Comment (wxtim) [reply src/user-guide/writing-workflows/jinja2.rst:251]:
```suggestion
- :ref:`Using a plugin`, such as :ref:`Cylc Rose`.
```

**Review by datamel — COMMENTED** [review]
Ah I see the error has changed now the tutorial is passing since the last push...
`Warning, treated as error:
/home/runner/work/cylc-doc/cylc-doc/src/user-guide/writing-workflows/jinja2.rst:251:Unknown target name: "using a plugin (example)`
This does look related to the change, I can reproduce this error locally.

**Comment by wxtim** [comment]
Sorry @MetRonnie  and @datamel  to leave this one stalled. I missed the new failure.

Now just failing on the tutorial workflow again!

**Review by MetRonnie — APPROVED** [review]

**Review by datamel — APPROVED** [review]
I have checked this out and built the docs locally. All working nicely, thanks @wxtim!

**Pull request merged**